service/service.go

```diff
@@ -195,22 +199,40 @@ func (s *Service) BackupGuestClusters() error {
 
 			TmpDir: tmpDir,
 		}
-		// run backup task
-		err = etcd.FullBackup(&backupConfig)
+
+		err = backupv3WithRetry(s.Retries, backupConfig, s.Logger)
 		if err != nil {
 			failed = true
 			s.Logger.Log("level", "error", "msg", "Failed to backup etcd cluster "+clusterID, "reason", err)
-			continue
 		}
-
 	}
+
 	// check if any backup failed
 	if failed {
-		time.Sleep(time.Minute * 5)
```

Comment:
Not sure why we need to wait for 5 minutes if the backup fails(It is a loop not a goroutine)

Reply:
Maybe it's to give some time for etcd to calm down and not the schedule next pod immediately and pressure it even more.

Reply:
Now we introduce 20 seconds wait every fail in 10 retries so if it is failing it would be waiting some time as well.
Do you think it could help having both of them?

Reply:
I can't tell what is better here.